crates/m3/src/builder/witness.rs

```diff
@@ -609,7 +616,7 @@ pub trait TableFiller<U: UnderlierType = OptimalUnderlier, F: TowerField = B128>
 	/// Fill the table witness with data derived from the given rows.
 	fn fill<'a>(
 		&'a self,
-		rows: impl Iterator<Item = &'a Self::Event>,
+		rows: impl Iterator<Item = &'a Self::Event> + Clone,
```

Comment:
I don't think adding `Clone` is necessary?

Reply:
I had to because I needed to clone the events when filling the cpu columns [in ZCray](https://github.com/IrreducibleOSS/zCrayVM/blob/c63d851780e6ddeb75536ece51868b3dac635c91/assembly/src/event/arithmetization/ret.rs#L132)